dive_core/command_hierarchy.cpp

```diff
@@ -755,6 +759,19 @@ bool CommandHierarchyCreator::OnIbStart(uint32_t                  submit_index,
         ib_string_stream << "DrawState IB"
                          << ", Address: 0x" << std::hex << ib_info.m_va_addr
                          << ", Size (DWORDS): " << std::dec << ib_info.m_size_in_dwords;
+
+        // Make all subsequent shared node parent be the relevant group in the set_draw_state packet
```

Comment:
sorry still a bit confused: what is shared node parent?

Reply:
A "shared node" is a node that can have multiple parents (i.e. shared among many different topologies). This is used exclusively for nodes that represent PM4 packets. PM4 packets can be children of IB-nodes, or children of other shared-nodes (e.g. children of INDIRECT_BUFFER packets), or children of draw call nodes, or even children of submit-nodes. So we needed to treat them a little differently.

Reply:
aha, thanks!